mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ui/ReceiptActivity.java

```diff
@@ -178,7 +178,7 @@ public void showTransferDetail(TransferDetail transferDetail) {
         tvTransFromName.setText(Constants.NAME + transferDetail.getFromClient().getDisplayName());
         tvTransFromNumber.setText(Constants.ACCOUNT_NUMBER + transferDetail
                 .getFromAccount().getAccountNo());
-        hideProgressDialog();
+        cancelProgressDialog();
```

Comment:
Why cancel it? We could have hiden it simply it _after_ the permission granting results

Reply:
Due to hiding the dialog didnt become null, so after the permission dialog the dialog was recreated repeatedly

Reply:
Sorry, I meant dismissing it. Canceling also makes a call to `onCancelListener` interface if I remember correctly. Simply dismissing after it's work is done seems better to me

Reply:
Yeah will dismiss it then and check if it still works